### PR TITLE
Add CapabilityStatementIndexer

### PR DESCRIPTION
**Pull request by JPercival**
* Add Indexer for CapabilityStatements
* Tests for same
* Update SearchParameterResolver to respect the index if present

### CODE REVIEW

File: engine.fhir/src/main/java/org/opencds/cqf/cql/engine/fhir/retrieve/RestFhirRetrieveProvider.java [hunk 1]
```diff
@@ -26,15 +29,32 @@
...
 		// client are on the same version of FHIR.
 		this.fhirClient = fhirClient;
-		this.searchStyle = DEFAULT_SEARCH_STYLE;
-	}
+        this.searchStyle = DEFAULT_SEARCH_STYLE;
+    }
+
+    @Override
+    protected synchronized CapabilityStatementIndex getIndex() {
+        if (this.getIndex() == null) {
```
Comment (csandersdev) [inline engine.fhir/src/main/java/org/opencds/cqf/cql/engine/fhir/retrieve/RestFhirRetrieveProvider.java:46]: We are getting a StackOverflowError from this line. It looks like a bug to me. The code should be this.index == null not this.getIndex().

**Review by csandersdev — CHANGES_REQUESTED** [review]
We are tracking 1.5.1-SNAPSHOT and our builds started failing today. I've marked the line of code that is causing the problem. Is it your expectation that a build on a feature branch would update the SNAPSHOTS in the repository? Seems like that should only happen on the develop branch. Not a huge deal, but FYI.

**Comment by msargentibm** [comment]
Related to the SNAPSHOT being updated from a feature branch, looks like that happened back when .travis.yml was updated to use bash  scripts. Before the change, deployments were only for develop and master:  https://github.com/DBCG/cql_engine/commit/1431958c2e031f6ff9df95bde17fa373f4cf9d57#diff-6ac3f79fc25d95cd1e3d51da53a4b21b939437392578a35ae8cd6d5366ca5485L25

After the change, deployments happen regardless of the branch in the bash script. Not sure if the behavior change was intentional, but wanted to point out that the behavior did used to be different.

**Comment by JPercival** [comment]
It should not be publishing from feature branches. I'll fix that.

**Comment by JPercival** [comment]
I reverted the SNAPSHOT to the develop branch and disabled the build on this branch until I have time to look at it. Busy at the Connectathon today. Give you builds a shot.

**Comment by JPercival** [comment]
This is going to rolled into a planning phase in the engine using a slightly different approach

**Pull request closed without merging**